bundles/org.openhab.binding.warmup/src/main/java/org/openhab/binding/warmup/internal/handler/MyWarmupAccountHandler.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.warmup.internal.handler;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.warmup.internal.api.MyWarmupApi;
+import org.openhab.binding.warmup.internal.api.MyWarmupApiException;
+import org.openhab.binding.warmup.internal.discovery.WarmupDiscoveryService;
+import org.openhab.binding.warmup.internal.model.query.QueryResponseDTO;
+
+/**
+ * @author James Melville - Initial contribution
+ */
+@NonNullByDefault
+public class MyWarmupAccountHandler extends BaseBridgeHandler {
+
+    private final MyWarmupApi api;
+
+    private @Nullable QueryResponseDTO queryResponse = null;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable WarmupDiscoveryService discoveryService;
+
+    public MyWarmupAccountHandler(Bridge thing, final HttpClient httpClient) {
+        super(thing);
+        api = new MyWarmupApi(httpClient, getConfigAs(MyWarmupConfigurationDTO.class));
+    }
+
+    @Override
+    public void initialize() {
+        MyWarmupConfigurationDTO config = getConfigAs(MyWarmupConfigurationDTO.class);
+        api.setConfiguration(config);
+        refreshJob = scheduler.scheduleWithFixedDelay(this::refreshFromServer, 0, config.refreshIntervalSec,
```

Comment:
Better validate the value of `config.refreshIntervalSec` as it can be 0 if the user doesn't configure it. Then, an IllegalArgumentException would be raised.

Reply:
Implemented guard against value being below 10 specified in thing.xml https://github.com/openhab/openhab-addons/pull/7998/commits/7bc9ed18881442a51c73f95fe123773f6d04e482#diff-028f0585e493ccac6de2540a9f878ba5